### PR TITLE
refactor: standardize googletest naming

### PR DESCRIPTION
**Pull request by bassmang**
Remove `_tests` from group and `test_` from individual tests to shorten filtering and consistency

### CODE REVIEW

File: vowpalwabbit/slim/test/ut_util.cc [hunk 1]
```diff
@@ -351,7 +351,7 @@ invalid_model_param invalid_model_param[] = {
 
 INSTANTIATE_TEST_SUITE_P(VowpalWabbitSlim, invalid_model_test, ::testing::ValuesIn(invalid_model_param));
 
-TEST(VowpalWabbitSlim, multiclass_data_4)
+TEST(vowpal_wabbit_slim, multiclass_data_4)
```
Comment (jackgerrits) [inline vowpalwabbit/slim/test/ut_util.cc:354]: If you change this you also need to update this: https://github.com/VowpalWabbit/vowpal_wabbit/blob/5c920b44f3f554820ff1fce01a7976dd1907814f/.github/workflows/build_vw_slim.yml#L31

File: vowpalwabbit/slim/test/ut_explore.cc [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 using namespace ::testing;
 
-TEST(ExploreTestSuite, sampling_rank)
+TEST(explore_slim, sampling_rank)
```
Comment (jackgerrits) [inline vowpalwabbit/slim/test/ut_explore.cc:12]: If you change this you also need to update this: https://github.com/VowpalWabbit/vowpal_wabbit/blob/5c920b44f3f554820ff1fce01a7976dd1907814f/.github/workflows/build_vw_slim.yml#L31

File: vowpalwabbit/slim/test/ut_opts.cc [hunk 3]
```diff
@@ -4,15 +4,15 @@
 using namespace ::testing;
 using namespace vw_slim;
 
-TEST(CommandLineOptions, parsing)
+TEST(command_line_options_slim, parsing)
```
Comment (jackgerrits) [inline vowpalwabbit/slim/test/ut_opts.cc:7]: If you change this you also need to update this: https://github.com/VowpalWabbit/vowpal_wabbit/blob/5c920b44f3f554820ff1fce01a7976dd1907814f/.github/workflows/build_vw_slim.yml#L31

**Review by jackgerrits — COMMENTED** [review]
I didn't realize this but according to the gtest docs the test suite name can't include `_`

https://google.github.io/googletest/reference/testing.html#TEST

While you're here it might be a good time to fix that up?

**Comment by bassmang** [comment]
> I didn't realize this but according to the gtest docs the test suite name can't include `_`
> 
> https://google.github.io/googletest/reference/testing.html#TEST
> 
> While you're here it might be a good time to fix that up?

Good catch, I'll update to camel case

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**